### PR TITLE
[tests] Add hot plug tests and change environment

### PR DESCRIPTION
**Pull request by pkulax**
Add more configure to dotenv
Add error handling
Change to platform as main construct
Add hot plug tests with minimal and maximal ramdrives Add negative test with above maximal ramdrives

Signed-off-by: Kula <przemyslawx.kula@intel.com>

### CODE REVIEW

File: build/storage/tests/it/ptf_tests/system_tools/errors.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""System tools errors"""
+
+
+class ContainerNotRunningException(Exception):
+    """Container is not currently running"""
+
+
+class CommandException(Exception):
+    """Custom Exception raises if error occurs during command execution"""
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/errors.py:9]: Custom exception was risen during ...

File: build/storage/tests/it/ptf_tests/system_tools/errors.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class VirtualizationException(Exception):
+    """Virtualization is not setting properly"""
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/errors.py:13]: Virtualization may not be set properly

File: build/storage/tests/it/ptf_tests/system_tools/errors.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class DependenciesException(Exception):
+    """Dependency is not setting properly"""
+
+
+class BusyPortException(Exception):
+    """There is process in port"""
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/errors.py:21]: The requested port is occupied by another process

File: build/storage/tests/it/ptf_tests/system_tools/errors.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class CommandException(Exception):
+    """Custom Exception raises if error occurs during command execution"""
+
+
+class VirtualizationException(Exception):
+    """Virtualization is not setting properly"""
+
+
+class DependenciesException(Exception):
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/errors.py:16]: DependencyException

File: build/storage/tests/it/ptf_tests/system_tools/errors.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    """Dependency is not setting properly"""
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/errors.py:17]: Provide a meaningful description for this generic exception

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 6]
```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
     def check_system_setup(self):
         """Overwrite this method in specific platform if you don't want check all setup"""
         if not self._is_virtualization():
-            raise Exception("Virtualization is not setting properly")
+            raise DependenciesException("Virtualization is not setting properly")
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:158]: You always return "DependenciesException" although you defined specialized classes for the different cases - use them here.

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 7]
```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
...
         if not self._is_kvm():
-            raise Exception("KVM is not setting properly")
+            raise DependenciesException("KVM is not setting properly")
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:160]: Use "is not set" instead of "is not setting" if formulating this way

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 8]
```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
...
+            raise DependenciesException("QUEMU is not setting properly")
         if not self._set_security_policies():
-            raise Exception("Security polices is not setting properly")
+            raise DependenciesException("Security polices is not setting properly")
         if not self._is_docker():
-            raise Exception("Docker is not setting properly")
+            raise DependenciesException("Docker is not setting properly")
 
-    # TODO: after testing restore settings
     def set_system_setup(self):
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:170]: Provide a more meaningful name and description - system_setup may be too generic. If on the other hand this method is supposed to cover all system setup related tasks, then provide a detailed description what it does.

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 9]
```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
...
+            f"sudo netstat -anop | grep -Po ':{port}\s.*LISTEN.*?\K\d+(?=/)' || true"
+        )
 
-class StorageTargetPlatform(BaseTestPlatform):
-    def __init__(self):
-        config = StorageTargetConfig()
-        terminal = SSHTerminal(config)
-        super().__init__(terminal)
+    def kill_process_from_port(self, port: int):
+        """Raise error if there is no process in specific port"""
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:182]: "occupying" instead of "in"

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 10]
```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
...
 
-    # TODO add implementation
-    def create_virtio_blk_device(self):
-        return "VirtioBlkDevice"
+    def delete_virtio_blk_devices(self, devices_handles):
+        for device_handle in devices_handles:
+            device_handle.delete(self.cmd_sender)
+
+    def clean(self):
+        # TODO delete all alocated devices
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:280]: is this not happening?

File: build/storage/tests/it/ptf_tests/system_tools/vm.py [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+from tenacity import retry, stop_after_delay
+
+from system_tools.const import (DEFAULT_HOST_TARGET_SERVICE_PORT_IN_VM,
+                                DEFAULT_QMP_PORT)
+from system_tools.errors import BusyPortException, VirtualizationException
+
+
+class VirtualMachine:
+    def __init__(self, platform):
+        # TODO: replace all these default parameters to values from config and change the way
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/vm.py:24]: with values

**Review by intelfisz — CHANGES_REQUESTED** [review]

File: build/storage/tests/it/ptf_tests/system_tools/errors.py (hunk 1 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/errors.py:9]: changed

File: build/storage/tests/it/ptf_tests/system_tools/errors.py (hunk 2 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/errors.py:13]: changed

File: build/storage/tests/it/ptf_tests/system_tools/errors.py (hunk 3 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/errors.py:21]: changed

File: build/storage/tests/it/ptf_tests/system_tools/errors.py (hunk 4 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/errors.py:16]: changed

File: build/storage/tests/it/ptf_tests/tests/test_env.py [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+        self.assertEqual(
+            self.host_target_terminal.execute("whoami"),
+            self.host_target_terminal.config.username,
+        )
+
+    def tearDown(self):
+        pass
+
+
+class TestTerminalConnectHasRootPrivilegnes(BaseTest):
```
[the remark this answers is not in the document]
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/tests/test_env.py:39]: changed

File: build/storage/tests/it/ptf_tests/tests/test_env.py [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+    def runTest(self):
+        self.storage_target_terminal.execute(
+            f"sudo rm -rf {self.storage_target_terminal.config.workdir}"
+        )
+        self.storage_target_terminal.execute(
+            f"mkdir {self.storage_target_terminal.config.workdir}"
+        )
+
+        # TODO: init submodules
```
[the remark this answers is not in the document]
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/tests/test_env.py:81]: in tests we don't need it so deleted comment

File: build/storage/tests/it/ptf_tests/tests/test_env.py [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+            "storage-target", self.storage_target_terminal.execute("docker ps")
+        )
+
+        self.ipu_storage_terminal.client.exec_command(f"mkdir -p {self.shared_dir}")
+        self.ipu_storage_container = IPUStorageContainer()
+        self.assertIn(
+            "ipu-storage-container", self.ipu_storage_terminal.execute("docker ps")
+        )
+
+        # container stops after few seconds
```
[the remark this answers is not in the document]
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/tests/test_env.py:103]: changed comment

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py (hunk 8 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/test_platform.py:170]: deleted - user should themselves set properly env

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py (hunk 9 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/test_platform.py:182]: changed

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py (hunk 10 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/test_platform.py:280]: now all containers are kill and but it nvme it is not enough

File: build/storage/tests/it/ptf_tests/system_tools/config.py [hunk 15]
```diff
@@ -14,55 +21,67 @@ class BaseConfig(ABC):
     def __init__(self):
         pass
 
+    def _getenv(self, env_name, alternative=None):
+        env = os.getenv(env_name)
+        return env if env else alternative
+
 
-class MainPlatformConfig(BaseConfig):
+class TestsConfig(BaseConfig):
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/config.py:29]: TestConfig

File: build/storage/tests/it/ptf_tests/system_tools/test_platform.py [hunk 16]
```diff
@@ -96,62 +155,165 @@ def _install_docker(self):
     def check_system_setup(self):
         """Overwrite this method in specific platform if you don't want check all setup"""
         if not self._is_virtualization():
-            raise Exception("Virtualization is not setting properly")
+            raise DependencyException("Virtualization is not setting properly")
```
Comment (intelfisz) [inline build/storage/tests/it/ptf_tests/system_tools/test_platform.py:158]: You still return the same exception type. Exception messages are the same as the ones which you corrected in errors.py

**Review by intelfisz — CHANGES_REQUESTED** [review]

File: build/storage/tests/it/ptf_tests/system_tools/config.py (hunk 15 above)
Comment (pkulax) [reply build/storage/tests/it/ptf_tests/system_tools/config.py:29]: changed

**Review by intelfisz — APPROVED** [review]
